vdom/core.py

```diff
@@ -125,13 +163,23 @@ def __init__(self, *children, **kwargs):
             self.children = _flatten_children(*children, **kwargs)
             self.attributes = kwargs
             self.tagName = tagName
+            self._schema = VDOM_SCHEMA
 
         def _repr_mimebundle_(self, include, exclude, **kwargs):
             return {
-                'application/vdom.v1+json': toJSON(self),
+                'application/vdom.v1+json': to_json(self, schema=self._schema),
                 'text/plain': '<{tagName} />'.format(tagName=tagName)
             }
 
+        @property
+        def schema(self):
+            return self._schema
+
+        @schema.setter
+        def schema(self, value):
+            Draft4Validator.check_schema(value)
+            self._schema = value
```

Comment:
Now that this has gotten bigger, should we make a `BaseComponent` that we inherit from for our created Component classes?

Reply:
I was thinking that but I didn't want to expand scope!

Reply:
Always room to do it after this PR.